src/docfx/lib/HtmlUtility.cs

```diff
@@ -165,24 +167,22 @@ public static string TransformXref(string html, Func<string, bool, int, (string?
                     continue;
                 }
 
-                var xref = HttpUtility.HtmlDecode(node.GetAttributeValue("href", ""));
+                var rawSource = HttpUtility.HtmlDecode(node.GetAttributeValue("data-raw-html", null) ?? node.GetAttributeValue("data-raw-source", null));
+                var isShorthand = rawSource?.StartsWith("@") ?? false;
 
-                var raw = HttpUtility.HtmlDecode(
-                    node.GetAttributeValue("data-raw-html", null) ?? node.GetAttributeValue("data-raw-source", null) ?? $"<span class=\"xref\">{HttpUtility.HtmlEncode(UrlUtility.SplitUrl(xref).path)}</span>");
+                var href = HttpUtility.HtmlDecode(node.GetAttributeValue("href", null));
+                var uid = href == null ? HttpUtility.HtmlDecode(node.GetAttributeValue("uid", null)) : default;
 
-                var isShorthand = raw.StartsWith("@");
-
-                var (resolvedHref, display) = transform(xref, isShorthand, columnOffset);
+                var (resolvedHref, display) = resolveXref(
+                    href == null ? null : (SourceInfo<string>?)new SourceInfo<string>(href, block?.ToSourceInfo(columnOffset: columnOffset)),
+                    uid == null ? null : (SourceInfo<string>?)new SourceInfo<string>(uid, block?.ToSourceInfo(columnOffset: columnOffset)),
+                    isShorthand);
 
                 var resolvedNode = new HtmlDocument();
                 if (string.IsNullOrEmpty(resolvedHref))
-                {
-                    resolvedNode.LoadHtml(raw);
-                }
+                    resolvedNode.LoadHtml(rawSource ?? $"<span class=\"xref\">{(href != null ? UrlUtility.SplitUrl(href).path : uid)}</span>");
```

Comment:
Consider leave the bracket. There was a stylecop rule for this but it was turned off.